docs/documentation/siddhi-4.0.md

```diff
@@ -1754,6 +1754,176 @@ define stream TempStream (deviceID long, roomNo int, temp double);
+
+
+!!! note
+    Table attributes must be always referred to with the table name as, 
+     `<table name>.<attibute name>`.
+
+**Example**
+
+The following query updates the room occupancy by increasing its value by 1, in the `RoomOccupancyTable` table for each room number greater than 10.
+
+```sql
+select 10 as roomNumber, 1 as arrival
+update RoomTypeTable
+    set RoomTypeTable.people = RoomTypeTable.people + arrival
+    on RoomTypeTable.roomNo == roomNumber;
+```
+
+### Update or Insert Into
+
+This allows you to update the given attributes, if a matching record to the given condition already exists in the database table.  
+Or else insert the entry as a new record.
+
+**Syntax**
+
+```sql
+select <attribute name>, <attribute name>, ...
+update or insert into <table>
+    set <table>.<attribute name> = <expression>, <table>.<attribute name> = <expression>, ...
+    on <condition>
+```
+The `condition` element specifies the basis on which records are selected for update.
+When specifying the `condition`, table attributes should be referred to with the table name.
+If a record that matches the condition does not already exist in the table, the arriving event is inserted into the table.
+
+The `set` clause is only used when an update is performed during the insert/update operation.
+When `set` clause is used, the attribute to the left is always a table attribute, and the attribute to the right can be a stream/table attribute, mathematical
+operation or other. The attribute to the left (i.e., the attribute in the event table) is updated with the value of the attribute to the right if the given condition is met. When the `set` clause is not provided, all the attributes in the table are updated.
+
+!!! note
+    Table attributes should be always referred to with the table name as `<table name>.<attibute name>`.
+
+**Example**
+
+The following query update the records in the `UpdateTable` table that have room numbers that match the same in the selection. If such records are found in the event table, they are updated. If such records are not found, it is inserted from the stream.
+
+```sql
+from RoomAssigneeStream
+select 10 as roomNo, "single" as type, "abc" as assignee
+update or insert into RoomAssigneeTable
+    set RoomAssigneeTable.assignee = assignee
+    on RoomAssigneeTable.roomNo == roomNo;
+```
+
+### Select Insert Into
+
+This allows you to insert a new record to the database table with the attribtue values you define in the select section.
+
+**Syntax**
+
+```sql
+select <attribute name>, <attribute name>, ...
+insert into <table>;
+```
+
+**Example**
+
+The following store query insret a new record to the table **RoomOccupancyTable**,  with the attribute values as below.
```

Comment:
**Suggestion**: "This store query inserts a new record to the `RoomOccupancyTable` table with the specified attribute values."

**General Comment**: This is the only description that starts with "This store query..." If the queries in the other examples are also store queries, we must start the description the same way for those as well. If not, we are implying that this is the only store query.